src/Context/FeatureContext.php

```diff
@@ -319,6 +324,57 @@ private static function get_behat_internal_variables() {
 		return $variables;
 	}
 
+	/**
+	 * Download and configure a copy of the SQLite plugin. This will add the db.php dropin to our cached
+	 * copy of WordPress files that are used for each scenario
+	 */
+	private static function download_and_configure_sqlite_plugin( $dir ) {
```

Comment:
Something that I just realized:

We need to split the _download_ and the _configure_ part into two separate methods.

The reason is that `cache_wp_files()` initializes this only once and then keeps things cached, which means subsequent test runs with different `DB_TYPE` will not work.

This currently does not work as the subsequent test run will break:

```bash
rm -rf $TMPDIR/wp-cli-test-*
composer behat # Downloads WP 6.3 and caches it _without SQLite_; runs all tests except `@require-sqlite`
DB_TYPE=sqlite composer behat # Does NOT download WP nor configure SQLite; runs `@require-sqlite` tests, which *fail* because SQLite is not configured
```

In `cache_wp_files()` we should just download & extract the SQLite ZIP file and store it somewhere. We can call this method `cache_sqlite_files()` or something

**Bonus:** use a different cache dir for the ZIP file than `self::$cache_dir`, as the ZIP file is not tied to a specific WP version like `self::$cache_dir` is. We can introduce `self::$sqlite_cache_dir` that just points to `sys_get_temp_dir() . '/wp-cli-test-sqlite-integration-cache'`

Then, in `download_wp()` we can copy the SQLite plugin to `wp-content/plugins` and copy the drop-in to the right place.

Hope my explanation makes sense, please let me know otherwise. Happy to help if needed!

Reply:
Sure, that makes sense! There already is some code right now to allow it to work on subsequent test runs with the same cache dir because I ran into exactly that problem while testing:

https://github.com/wp-cli/wp-cli-tests/pull/181/files#diff-2347198287ada7324ce7ede2aff44e5a4445c493fc0059f8c0b5f55f3461ecb8R387-R395

So it will always download sqlite to the existing cache dir if `DB_TYPE=sql` or remove it if `DB_TYPE=mysql`. It can probably be better though and I'll try and implement those suggestions

Reply:
OK I was able to make the suggested changes. There is a now a separate download and configure step, and the download is cached in its own `sys_get_temp_dir() . '/wp-cli-test-sqlite-integration-cache'` folder similar to WP core files.